File: components/content_creation/notes/android/java/src/org/chromium/components/content_creation/notes/models/NoteTemplate.java
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2021 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.components.content_creation.notes.models;

/**
 * Model class for note templates.
 */
public class NoteTemplate {
    /** The localized name of this template. */
    public final int id;
    public final String localizedName;
    public final Background mainBackground;
    public final TextStyle textStyle;
    public final FooterStyle footerStyle;

    /** Constructor. */
    public NoteTemplate(int id, String localizedName, Background mainBackground,
            TextStyle textStyle, FooterStyle footerStyle) {
        this.id = id;
        this.localizedName = localizedName;
        this.mainBackground = mainBackground;
        this.textStyle = textStyle;
        this.footerStyle = footerStyle;
    }
}

File: components/content_creation/notes/core/templates/note_template.cc
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2021 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "components/content_creation/notes/core/templates/note_template.h"

namespace content_creation {

NoteTemplate::NoteTemplate(const std::string& localized_name)
    : localized_name_(localized_name) {}

}  // namespace content_creation
